x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/details/execution_log_table/rule_duration_format.tsx

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import numeral from '@elastic/numeral';
+import moment from 'moment';
+import React, { useMemo } from 'react';
+
+interface Props {
+  duration: number;
+  isMillis?: boolean;
+  allowZero?: boolean;
+}
+
+export function getFormattedDuration(value: number) {
```

Comment:
Is there a platform utility for formatting durations as design mocked (hh:mm:ss:SSS)? Didn't see anything exported, so adapted from an existing alerting component.

Reply:
There are not as far as I've found. I've expanded this component's functionality and test coverage in https://github.com/elastic/kibana/pull/130072 to support `days` and duration overflow.